### PR TITLE
WIP: add recaptcha widget and verification for guest registration

### PR DESCRIPTION
**Pull request by cguardia**
@davisagli I still haven't finished this, but I wanted to see if you have any comments so far. Having a bit of trouble with the flow after hitting 'register', but the widget itself works.

### CODE REVIEW

**Comment by davisagli** [comment]
I think what you're missing is that the button calls `handleSubmit`, which comes from the FormMixin and is different from `handleSubmitted`, which is a callback that the mixin calls *after* it's received a successful response from the backend.

So I think what you need to do is:
1. Rename the code you currently have in handleSubmitted and make the button call that instead of `handleSubmit`
2. Make `validateCaptcha` call `handleSubmit` once the captcha is verified
3. rename `nextStep` back to `handleSubmitted` so it gets called by the form mixin.

File: src/vms/static/app/register.js [hunk 1]
```diff
@@ -521,6 +522,16 @@ var RegisterForm = React.createClass({
 					: '' }
 					<hr />
 					<br />
+                    { !this.props.user ?
+                        <div>
+                          <h2>Sentient Life Verification</h2>
+                          <Recaptcha sitekey="6LcHaw4TAAAAAMFZadjGQjaa-qqwXwqDUs0q6UL0" />
```
Comment (davisagli) [inline src/vms/static/app/register.js]: We should get the site key from an environment variable too. This is a public repository.

File: src/vms/templates/main_template.pt [hunk 2]
```diff
@@ -7,6 +7,7 @@
 	<title>${layout.site_title}</title>
 	<link rel="stylesheet" href="${request.cachebusted_path('vms:static/css/style.css')}">
 	<base href="${request.resource_path(context)}" tal:on-error="nothing" />
+    <script src='https://www.google.com/recaptcha/api.js'></script>
```
Comment (davisagli) [inline src/vms/templates/main_template.pt]: Please put it at the bottom with the other scripts.

**Comment by cguardia** [comment]
You are right, that's what I missed. It's so obvious now ;)

Thanks.

**Comment by cguardia** [comment]
@davisagli this is almost ready, but I'm not sure how to pass the event to handleSubmit after the captcha is verified (I get "e is null"). Suggestions?

**Comment by davisagli** [comment]
I'm not looking at the code now, but can you adjust handleSubmit so it also works without an event? What does it use the event for?

David Glick
(mobile)

**Comment by cguardia** [comment]
e.preventDefault(), but removing that would affect other uses of the form mix-in, I fear.

**Comment by davisagli** [comment]
So do something like this:

    if (e !== undefined) e.preventDefault();

and then it can be used both ways.

**Comment by cguardia** [comment]
Yeah, thanks. Guess, I was being a bit stubborn here about forwarding the event.

**Comment by davisagli** [comment]
Yeah, I'm not quite sure why just passing the event doesn't work.

**Comment by cguardia** [comment]
@davisagli I made that change, but now I'm a bit confused. I merged from master and now you can't register without a backpack account, so this feature would never be used. Am I missing something?

**Comment by davisagli** [comment]
> now you can't register without a backpack account

You can still register as a guest for work party types with "Allow Guest Registration" checked

**Comment by cguardia** [comment]
Is that the "Group WP open to public" flag? Don't see anything else in the SF UI, but this doesn't do anything. I still see the backpack registration required message.

**Comment by davisagli** [comment]
No, it's the "Allow Guest Registrations" field on the Work Party Type object. Try the "David Test" work party; I have it turned on for that one.

**Comment by cguardia** [comment]
"David Test" also forces registration. I tried every single WP and none allows anonymous registration.

**Comment by davisagli** [comment]
Aha, there was a merge error when @alecpm refactored how the work party query is built and merged it with @cewing's change to include the Allow_Guest_Registrations__c field, and that field was no longer getting fetched. I fixed it.

**Comment by alecpm** [comment]
Sorry :-(

**Comment by cguardia** [comment]
@davisagli I think this is finally ready.

**Pull request merged**